### ISSUE TITLE
Failure to decode DNS response

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#919]
**Describe the bug**
Decoding of something received when scanning causes a suppressed error. I added an exception log to simplify.

**To Reproduce**
Just scan (here with exception log);

```
$  atvremote --debug --mdns-debug scan
...
2021-01-09 22:15:43 ERROR: bad stuff
Traceback (most recent call last):
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/mdns.py", line 337, in datagram_received
    services = parse_services(DnsMessage().unpack(data))
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 291, in unpack
    self.answers.extend(
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 292, in <genexpr>
    DnsResource.unpack_read(buffer) for _ in range((header.ancount))
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 258, in unpack_read
    rd = qtype.parse_rdata(buffer, rd_length)
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 173, in parse_rdata
    return parse_domain_name(buffer)
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 108, in parse_domain_name
    return ".".join(label.decode("idna") for label in labels)
  File "/home/postlund/pyatv_dev/pyatv/pyatv/support/dns.py", line 108, in <genexpr>
    return ".".join(label.decode("idna") for label in labels)
  File "/usr/local/lib/python3.8/encodings/idna.py", line 214, in decode
    result.append(ToUnicode(label))
  File "/usr/local/lib/python3.8/encodings/idna.py", line 125, in ToUnicode
    return str(label, "ascii")
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 5: ordinal not in range(128)
2021-01-09 22:15:43 DEBUG: Failed to decode message (Msg=35ff840000030003000000000b5f6170706c6574762d7632045f746370056c6f63616c00000c80010b5f746f7563682d61626c65c018000c8001085f616972706c6179c018000c8001c028000c00010000000a00131032444632453735303736333943323639c028c00c000c00010000000a00131032444632453735303736333943323639c00cc03a000c00010000000a0013104170706c65c2a0545620283431363729c03a)
...
```
Payload at the bottom is what fails to decode.

**Expected behavior**
Should hopefully decode correctly?

**System Setup (please complete the following information):**
 - OS: any
 - Python: 3.8.1
 - pyatv: `master`
- Apple TV: N/A

**Additional context**
@paxswill Maybe something for you? 😉

**Comment by paxswill** [comment]
Can I just say how much I dislike how permissive DNS-SD is for Service Instance Names? Instance names (the part before `_http._tcp`) can be practically anything that can be encoded as UTF-8, **including dots**. And the entire name (Instance, Service, Domain) doesn't conform to hostname restrictions either, so commas and spaces are alright. `Building 2, 1st Floor  .  example  .  com  .` is literally an example of a domain from the RFC.

So now my Apple TV has katakana and a dot in its name. Hopefully I'll have a good fix for this in a bit.

**Comment by postlund** [comment]
Be my guest! 🤣 I guess you should be able to reproduce this quite simple with a unit test as well?

**Comment by paxswill** [comment]
Yep, easy enough to reproduce for a unit test as well .Just thinking over how best to approach it.

**Comment by postlund** [comment]
May I suggest duct tape?

**Comment by postlund** [comment]
Fixed by #927.
